### PR TITLE
feat: Add Post Like management for all logged in users

### PR DESCRIPTION
**Pull request by Naman-Saxena1**
# Added Frontend and Backend for post like management
1. All logged in users can like posts
2. A record of all liked posts by user is maintained in MongoDb using ref
3. Like count is maintained for all posts in the db
4. Button displays as "Like" or "Liked" depending upon if user has not liked post or has already liked post.

# Files to review:
1. UserPost.jsx
2. Home.jsx
3. Majority of changes are on backend so only these 2 files and can ignore Redux files.

# Login Creds to test:
## User 1
Email - newuser1@gmail.com
Password - zxcv

## User 2
Email - newuser2@gmail.com
Password - zxcv

### CODE REVIEW

**Comment by mehulsatardekar** [comment]
Just UI suggestion, the what's happening section side should remain sticky in nature.  the Rest of thing looks good @Naman-Saxena1

File: src/components/UserPost/UserPost.jsx [hunk 1]
```diff
@@ -88,9 +168,34 @@ function UserPost({userPostDetails})
...
+                        </button>
+                    ) : (
+                        <button 
+                            className="outline-secondary-btn post-options-bottom-buttons"
+                            onClick={() => handleOptionsHandler(handlePostLikeStatus)}
+                        >
+                            <AiOutlineLike className="post-options-button-icons"/>
+                            Like
+                        </button>
+                    )
```
Comment (bhtibrewal) [inline src/components/UserPost/UserPost.jsx:189]: suggestion: I would suggest you break down these buttons into separate components for better readability
as you are conditionally rendering them here

**Review by bhtibrewal — APPROVED** [review]
clean and understandable code, good work Naman, liked UI also, you can merge

**Review by nikhiljugale007 — APPROVED** [review]
Did not find any issue.
Ui is working properly. 
Clean and readable code. 
Good to merge.

**Pull request merged**